src/urllib3/response.py

```diff
@@ -944,6 +948,10 @@ def read(
                 # For example, the GZ file header takes 10 bytes, we don't want to read
                 # it one byte at a time
                 data = self._raw_read(amt)
+
+                # TODO verify if flush_decoder is redundant here, as it is not updated
+                #      anyways within the loop. Passing 'False' directly does not lead
+                #      to test-failures. Should be 'True' only(!) for last time called
```

Comment:
I identified a potential bug correctly here (fixed later on within https://github.com/urllib3/urllib3/pull/3262/files#r1460238172), but failed to update my report (https://github.com/urllib3/urllib3/issues/2799#issuecomment-1872448595, section `2023-12-27: Todays Code`)